src/main/java/org/dependencytrack/policy/ComponentHashPolicyEvaluator.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of Dependency-Track.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.dependencytrack.policy;
+
+import alpine.common.logging.Logger;
+import org.apache.commons.lang3.StringUtils;
+import org.cyclonedx.model.Hash;
+import org.dependencytrack.model.*;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Evaluates a components HASH against a policy.
+ */
+public class ComponentHashPolicyEvaluator extends AbstractPolicyEvaluator {
+
+    private static final Logger LOGGER = Logger.getLogger(ComponentHashPolicyEvaluator.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PolicyCondition.Subject supportedSubject() {
+        return PolicyCondition.Subject.COMPONENT_HASH;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<PolicyConditionViolation> evaluate(final Policy policy, final Component component) {
+        final List<PolicyConditionViolation> violations = new ArrayList<>();
+        for (final PolicyCondition condition : super.extractSupportedConditions(policy)) {
+            LOGGER.debug("Evaluating component (" + component.getUuid() + ") against policy condition (" + condition.getUuid() + ")");
+            final Hash hash = extractHashValues(condition);
+            if (matches(hash, component)) {
+                violations.add(new PolicyConditionViolation(condition, component));
+            }
+        }
+        return violations;
+    }
+
+    private Hash extractHashValues(PolicyCondition condition) {
+
+        if (condition.getValue() == null) {
+            return null;
+        }
+        final JSONObject def = new JSONObject(condition.getValue());
+        return new Hash(
+                def.optString("algorithm", null),
+                def.optString("value", null)
+        );
+    }
+
+    private boolean matches(Hash hash, Component component) {
+
+        if (hash == null)
+            return true;
```

Comment:
Similarly to our discussion in #1773, shouldn't the condition not *not* match when `hash` is null?